From fc65b0ca53916ed634b038ac1d0485bf2f4d1959 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:17:42 -0400
Subject: [alsa] hda: Add subwoofer quirks for Asus UX51VZH and N55SF

Message-id: <140972823573-91-patchset-jkysela@redhat.com>
Patchwork-id: 90791
O-Subject: [RHEL 7.1 PATCH 090/442] ALSA: hda - Add subwoofer quirks for Asus UX51VZH and N55SF
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Bug reporter report that the -mode4 makes the subwoofer work.
I have simplified the quirk a bit to avoid possible regressions
with the microphones.

BugLink: https://bugs.launchpad.net/bugs/871808
Signed-off-by: David Henningsson <david.henningsson@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 61a75f138a351baf0fdc8c210cebd06b9ced4279)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 7e59a7e..7a628e8 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -4927,6 +4927,7 @@ enum {
  ALC668_FIXUP_DELL_MIC_NO_PRESENCE,
  ALC668_FIXUP_HEADSET_MODE,
  ALC662_FIXUP_BASS_MODE4_CHMAP,
+ ALC662_FIXUP_BASS_16,
  ALC662_FIXUP_BASS_1A,
  ALC662_FIXUP_BASS_CHMAP,
  ALC668_FIXUP_AUTO_MUTE,
@@ -5122,6 +5123,15 @@ static const struct hda_fixup alc662_fixups[] = {
   .chained = true,
   .chain_id = ALC662_FIXUP_ASUS_MODE4
  },
+ [ALC662_FIXUP_BASS_16] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = (const struct hda_pintbl[]) {
+   {0x16, 0x80106111}, /* bass speaker */
+   {}
+  },
+  .chained = true,
+  .chain_id = ALC662_FIXUP_BASS_CHMAP,
+ },
  [ALC662_FIXUP_BASS_1A] = {
   .type = HDA_FIXUP_PINS,
   .v.pins = (const struct hda_pintbl[]) {
@@ -5157,6 +5167,8 @@ static const struct snd_pci_quirk alc662_fixup_tbl[] = {
  SND_PCI_QUIRK(0x103c, 0x1632, "HP RP5800", ALC662_FIXUP_HP_RP5800),
  SND_PCI_QUIRK(0x1043, 0x11cd, "Asus N550", ALC662_FIXUP_BASS_1A),
  SND_PCI_QUIRK(0x1043, 0x1477, "ASUS N56VZ", ALC662_FIXUP_BASS_MODE4_CHMAP),
+ SND_PCI_QUIRK(0x1043, 0x15a7, "ASUS UX51VZH", ALC662_FIXUP_BASS_16),
+ SND_PCI_QUIRK(0x1043, 0x1b73, "ASUS N55SF", ALC662_FIXUP_BASS_16),
  SND_PCI_QUIRK(0x1043, 0x1bf3, "ASUS N76VZ", ALC662_FIXUP_BASS_MODE4_CHMAP),
  SND_PCI_QUIRK(0x1043, 0x8469, "ASUS mobo", ALC662_FIXUP_NO_JACK_DETECT),
  SND_PCI_QUIRK(0x105b, 0x0cd6, "Foxconn", ALC662_FIXUP_ASUS_MODE2),
-- 
1.7.1